ui/src/main/webapp/src/app/components/rules-modal.component.ts

```diff
@@ -15,5 +17,14 @@ export class RulesModalComponent {
 
     show():void {
         (<any>$('#rulesModal')).modal();
+        // I can't figure out how to make this call prettyPrint() when it's shown.
+        prettyPrint(); // This is too soon, it's not rendered yet.
+        setTimeout(() => prettyPrint(), 1000);
```

Comment:
It is just an idea, but maybe you could try `ngAfterViewInit` lifecycle hook. 
https://angular.io/docs/ts/latest/guide/lifecycle-hooks.html